fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -1588,8 +1588,10 @@ public List<LoanTransaction> retrieveListOfTransactionsForReprocessing() {
     }
 
     private static Predicate<LoanTransaction> loanTransactionForReprocessingPredicate() {
-        return transaction -> transaction.isNotReversed() && (transaction.isChargeOff() || transaction.isReAge()
-                || transaction.isAccrualActivity() || transaction.isReAmortize() || !transaction.isNonMonetaryTransaction());
+        return transaction -> (transaction.isNotReversed() && (transaction.isChargeOff() || transaction.isReAge()
+                || transaction.isAccrualActivity() || transaction.isReAmortize() || !transaction.isNonMonetaryTransaction()))
+                || (transaction.isReversed() && transaction.isChargeOff() && LoanChargeOffBehaviour.ZERO_INTEREST
```

Comment:
This changes are not needed! Please undo them

Reply:
done